docs/data/date-pickers/base-concepts/ReferenceDateExplicitDateTimePicker.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+
+export default function ReferenceDateExplicitDateTimePicker() {
+  const [value, setValue] = React.useState<Dayjs | null>(null);
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={2}>
```

Comment:
Maybe we should use the `DemoContainer` and/or `DemoItem` here to avoid having a more narrow demo as compared to others, where the placeholder is cut off..? 🤔

Reply:
Or maybe set an explicit width such that it wouldn't require a layout shift when a value is selected?
Same question then applies to the TimePicker demo below.

Reply:
We have quite a lot of demos with 1 picker / field + one typography to display the output.
Maybe we could improve how they are handled to unify the width with `DemoContainer`.

I added fixed width for this PR